### PR TITLE
[Needs Review] Clean IStorage and enforce inheritance

### PR DESCRIPTION
**Pull request by polsm91**
- Dynamically add or remove the delete/stop persistent methods
- Enforce base IStorage structure when new persistent objects are created (by means of abc class and abstract methods)
- Remove unrelated methods from IStorage, or methods related to Cassandra
- Read stored metadatas when recreating objects from Storage_id
- UUIDs are instances of the UUID Class

### CODE REVIEW

File: hecuba_py/hecuba/qbeast.py [hunk 1]
```diff
@@ -114,33 +112,53 @@ def __init__(self, primary_keys, columns, indexed_on, name, qbeast_meta=None, qb
...
             class_name,
-            built_remotely)
+            self._built_remotely)
 
-        persistent_columns = [{"name": col[0], "type": col[1]} if isinstance(col, tuple) else col for col in columns]
+        if name or storage_id:
+            self.make_persistent(name)
+
+    def make_persistent(self, name):
+        if self._is_persistent:
```
Comment (cugni) [inline hecuba_py/hecuba/qbeast.py]: remove this if

File: hecuba_py/hecuba/qbeast.py [hunk 2]
```diff
@@ -169,3 +187,11 @@ def __iter__(self):
                                       self)
 
         return iterator
+
+    def _stop_persistent(self):
```
Comment (cugni) [inline hecuba_py/hecuba/qbeast.py]: remove this

File: hecuba_py/hecuba/qbeast.py [hunk 3]
```diff
@@ -169,3 +187,11 @@ def __iter__(self):
...
+        vars(self)['_stop_persistent'] = vars(self).pop('stop_persistent')
+        self.storage_id = None
+
+    def _delete_persistent(self):
```
Comment (cugni) [inline hecuba_py/hecuba/qbeast.py]: and remove this

File: hecuba_py/hecuba/IStorage.py [hunk 4]
```diff
@@ -1,262 +1,127 @@
...
-        tmap = tm.tokens_to_hosts_by_ks[ksp]
 
-    tokens_per_node = defaultdict(list)
-    for tmumur, t_to in tokens_murmur3:
-        point = bisect_right(tm.ring, tmumur)
-        if point == len(tm.ring):
-            tokens_per_node[tmap[tm.ring[0]][0]].append((tmumur.value, t_to))
-        else:
-            tokens_per_node[tmap[tm.ring[point]][0]].append((tmumur.value, t_to))
+    @abstractmethod
```
Comment (cugni) [inline hecuba_py/hecuba/IStorage.py]: remove abstractmethod, it is not always useful.

File: hecuba_py/hecuba/IStorage.py [hunk 5]
```diff
@@ -1,262 +1,127 @@
...
+    def _stop_persistent(self):
+        vars(self)['_stop_persistent'] = vars(self).pop('stop_persistent')
```
Comment (cugni) [inline hecuba_py/hecuba/IStorage.py]: del  vars(self)['_stop_persistent']

File: hecuba_py/hecuba/IStorage.py [hunk 6]
```diff
@@ -1,262 +1,127 @@
...
 
-def process_path(module_path):
-    """
-    Method to obtain module and class_name from a module path
-    Args:
-        module_path(String): path in the format module.class_name
-    Returns:
-        tuple containing class_name and module
-    """
+    @abstractmethod
```
[the remark this answers is not in the document]
Comment (polsm91) [reply hecuba_py/hecuba/IStorage.py]: In the end I opted to define a public "make/stop/delete persistent" and then, if the subclass wants to add functionality define a "_make_persistent"

**Pull request merged**